### ISSUE TITLE
Operator deployment should honor ROOK_UNREACHABLE_NODE_TOLERATION_SECONDS itself

### ISSUE THREAD

**Issue by jsalatiel** [rook/rook#12660]
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**

If one full zone is lost and the operator is running on that zone, it will take node-monitor-grace-period + 5 minutes by default to the operator be redeployed. There is a ROOK_UNREACHABLE_NODE_TOLERATION_SECONDS but that does not seem apply to the operator itself, only for "mgr, rbd, mds, rgw, nfs, PVC based mons and osds, and ceph toolbox"

**What is use case behind this feature:**
The operator should be quickly redeployed to do its job.

**Comment by travisn** [comment]
The `ROOK_UNREACHABLE_NODE_TOLERATION_SECONDS` setting is for the operator to add the toleration to all the ceph pods, but it doesn't apply to itself. So you could add the toleration to the operator deployment as seen in [this toleration](https://github.com/rook/rook/blob/a54c3cb415490fcdfa98ef6b8c1af2051c1cfc25/pkg/operator/k8sutil/pod.go#L144-L147). 

Ultimately, the operator could also consider running in HA mode as suggested in #701, but that would be a big work item.

**Comment by jsalatiel** [comment]
I've added it manually .
I opened this ticket because maybe defaulting to 5s on the operator deployment would be good for everyone.

**Comment by travisn** [comment]
True, at least it will start quicker elsewhere, at least if/until we implement the HA

**Comment by carlory** [comment]
Can I pick up it?

**Comment by subhamkrai** [comment]
We I have barely start working but I'll assign it to you @carlory

**Comment by travisn** [comment]
Unassigning due to inactivity

**Comment by travisn** [comment]
Note that the suggested fix would be to add the following toleration to [operator.yaml](https://github.com/rook/rook/blob/master/deploy/examples/operator.yaml) and [operator-openshift.yaml](https://github.com/rook/rook/blob/master/deploy/examples/operator-openshift.yaml):

```yaml
      tolerations:
      - effect: NoExecute
        key: node.kubernetes.io/unreachable
        operator: Exists
        tolerationSeconds: 5
```
